jest-puppeteer.config.js

```diff
@@ -30,7 +30,18 @@ module.exports = {
        */
       '--no-startup-window'
     ],
-    headless: process.env.HEADLESS !== 'false',
+
+    /**
+     * Allow headless mode switching using `HEADLESS=false`
+     * but default to `undefined` to skip deprecation warning
+     *
+     * {@link https://developer.chrome.com/articles/new-headless/}
+     */
+    headless: process.env.HEADLESS !== 'false'
+      ? undefined
```

Comment:
By defaulting to `{ headless: undefined }` we skip the deprecation checks in:

* https://github.com/puppeteer/puppeteer/pull/10039

Reply:
@domoscargin Sadly they're using [`const headless = options.headless ?? true;`](https://github.com/puppeteer/puppeteer/pull/10039/files#diff-61c2acef7c3c76385e015ae62bb14ba8789889e19dc57f7993a5dec0cb253ac2R48) which compiles to:

```js
const headless = (_a = options.headless) !== null && _a !== void 0 ? _a : true;
```

But we can default to `1` instead of `undefined` to make this work